src/main.c

```diff
@@ -104,8 +104,14 @@ struct listener_ssl_config_t {
 
 struct listener_config_t {
     int fd;
+#if defined(__linux__) && defined(SO_REUSEPORT)
+    int domain;
+    int so_reuseport;
+    H2O_VECTOR(int) reuseport_fds;
```

Comment:
How about changing the type to `int *`?

If set to NULL, it will indicate that `SO_REUSEPORT` is not used.
If set to non-NULL, it will be an array of `num_threads` entries containing the file descriptors.

By changing as such, we could omit assertions like `assert(listener->reuseport_fds.capacity == conf.num_threads)`.

Reply:
fixed.